Source/FerroX.cpp

```diff
@@ -411,6 +422,26 @@ void InitializeFerroXNamespace(const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM
      plot_int = -1;
      pp.query("plot_int",plot_int);
 
+
+     //hardswitch flag
+     hardswitch_flag = 0;
+     pp.query("hardswitch_flag",hardswitch_flag);
+
+     //hardswitch ratio
+     hardswitch_ratio = 0.05;
+     pp.query("hardswitch_ratio",hardswitch_ratio);
```

Comment:
It's better to have an if condition here:
```suggestion
if (hardswitch_flag == 1){
     pp.get("hardswitch_ratio",hardswitch_ratio);}
    else {
    amex::Abort("Hard to switch defects turned on but volume percentage undefined!");} 
     
```

Reply:
this logic will not work, change it to:
    if (hardswitch_flag == 1) {
        if (!pp.query("hardswitch_ratio", hardswitch_ratio)) {
            amrex::Abort("Hard to switch defects turned on but hardswitch_ratio is undefined!");
        }
    }